guides-and-tutorials/running-a-signer/signer-quickstart.md

```diff
@@ -0,0 +1,435 @@
+
+# Install needed packages
+sudo apt install -y npm wget unzip jq tar
+
+# Install Stacks CLI globally
+npm install --global @stacks/cli
+
+# Generate a new account and store details in a file
+# '-t' option makes this a testnet account
+stx make_keychain -t | jq > ~/stacks-signer/keychain.json
+```
+
+{% endtab %}
+
+{% endtabs %}
+
+The account file previously created looks like this:
+
+```json
+{
+  "mnemonic": "aaa bbb ccc ddd ...",
+  "keyInfo": {
+    "privateKey": "65f3...",
+    "publicKey": "03a3...",
+    "address": "SP1G...",
+    "btcAddress": "19tg...",
+    "wif": "Kzdt...",
+    "index": 0
+  }
+}
+```
+
+From this file, you'll need the `privateKey` value.
+
+## Step 2 - Set Up Your Stacks Signer
+
+### Download the stacks-signer binary
+
+Official binaries are available from the [Stacks Core releases page on Github](https://github.com/stacks-network/stacks-core/releases). Each release includes pre-built binaries. Download the [latest signer release ZIP file](https://github.com/stacks-network/stacks-core/releases/tag/signer-3.0.0.0.0.0) for your server’s architecture and decompress it. Inside of that folder is a `stacks-signer` binary.
+
+Assuming a `Linux x64 glibc` machine, the commands to download and uncompress the signer binary look like this:
+
+```bash
+# Enter the signer directory
+cd ~/stacks-signer
+
+# Download the signer binary zip
+wget https://github.com/stacks-network/stacks-core/releases/download/signer-3.0.0.0.0.0/linux-glibc-x64.zip
+
+# Unzip the signer binary archive
+unzip linux-glibc-x64.zip
+```
+
+### Create the configuration file
+
+Create the configuration file required to start the signer (be sure to replace `<your_token>` and `<your_private_key>` with your auth token and private key values):
+
+{% tabs %}
+
+{% tab title="Mainnet" %}
+
+```bash
+# Set environment variables
+AUTH_TOKEN=<your_token> # Used for signer-node authentication
+PRIVATE_KEY=<your_private_key> # privateKey from Step 1, this is the signer's private key
+
+# Create the signer's configuration file
+cat <<EOF> ~/stacks-signer/signer-config.toml
+node_host = "127.0.0.1:20443"
+endpoint = "127.0.0.1:30000"
+network = "mainnet"
+db_path = "$HOME/stacks-signer/data/signer.sqlite"
+auth_password = "$AUTH_TOKEN"
+stacks_private_key = "$PRIVATE_KEY"
+metrics_endpoint = "127.0.0.1:9154"
+block_proposal_timeout_ms = 180000
+EOF
+```
+
+{% endtab %}
+
+{% tab title="Testnet" %}
+
+```bash
+# Set environment variables
+AUTH_TOKEN=<your_token> # Used for signer-node authentication
+PRIVATE_KEY=<your_private_key> # privateKey from Step 1, this is the signer's private key
+
+# Create the signer's configuration file
+cat <<EOF> ~/stacks-signer/signer-config.toml
+node_host = "127.0.0.1:20443"
+endpoint = "127.0.0.1:30000"
+network = "testnet"
+db_path = "$HOME/stacks-signer/data/signer.sqlite"
+auth_password = "$AUTH_TOKEN"
+stacks_private_key = "$PRIVATE_KEY"
+metrics_endpoint = "127.0.0.1:9154"
+block_proposal_timeout_ms = 180000
+EOF
+```
```

Comment:
since the file with the key is being saved, we can entirely do away with the hard-code env vars. 
ex:
```
$ cat test.json 
{
  "privateKey":"bar"
}
$ cat <<EOF> ./test.toml
node_host = "127.0.0.1:30000"
stacks_private_key = "$(jq -r  '.privateKey' test.json)
auth_password = "$(openssl rand -base64 12)"
EOF
$ cat test.toml
node_host = "127.0.0.1:30000"
stacks_private_key = "bar
auth_password = "8ClPmSsvGQ/w+MBo"
```

Reply:
not sure which would be easier for the target audience, but it's an option

Reply:
@aldur opinion here?

Reply:
Optimizing for making it as easy as possible for users, I think using a sub shell to get the value of `stacks_private_key` might be confusing, even though it's a great idea:

I'd do something in the middle:

```bash
# Set environment variables
AUTH_TOKEN="$(openssl rand -base64 12)"  # Used for signer-node authentication
PRIVATE_KEY="$(jq -r  '.privateKey' test.json)" # or privateKey from Step 1, this is the signer's private key

# Now create the signer's configuration file
cat <<EOF> ~/stacks-signer/signer-config.toml
node_host = "127.0.0.1:20443"
endpoint = "127.0.0.1:30000"
network = "testnet"
db_path = "$HOME/stacks-signer/data/signer.sqlite"
auth_password = "$AUTH_TOKEN"
stacks_private_key = "$PRIVATE_KEY"
metrics_endpoint = "127.0.0.1:9154"
block_proposal_timeout_ms = 180000
EOF
```

Reply:
let's leave as-is for now then